### PR TITLE
cmd, core: extend freezer info dump for sub-ancient-store

### PR DESCRIPTION
**Pull request by rjl493456442**

### CODE REVIEW

File: cmd/geth/dbcmd.go [hunk 1]
```diff
@@ -537,41 +537,51 @@ func dbDumpTrie(ctx *cli.Context) error {
...
-		sort.Strings(options)
-		return fmt.Errorf("Could read freezer-type '%v'. Available options: %v", kind, options)
-	} else {
-		disableSnappy = noSnap
+
+	exist, disableSnappy, subdir := rawdb.FreezerTableInfo(kind)
+	if !exist {
+		tables := rawdb.FreezerTables()
+		sort.Strings(tables)
+		return fmt.Errorf("could read freezer-type '%v'. Available options: %v", kind, tables)
```
Comment (holiman) [inline cmd/geth/dbcmd.go:552]: I know it's not your error, but might as well fix it now: 
```suggestion
		return fmt.Errorf("could not read freezer-type '%v'. Available options: %v", kind, tables)
```

File: cmd/geth/dbcmd.go [hunk 2]
```diff
@@ -537,41 +537,51 @@ func dbDumpTrie(ctx *cli.Context) error {
...
 	}
 	if start, err = strconv.ParseInt(ctx.Args().Get(1), 10, 64); err != nil {
-		log.Info("Could read start-param", "error", err)
+		log.Info("Could read start-param", "err", err)
```
Comment (holiman) [inline cmd/geth/dbcmd.go:555]:
```suggestion
		log.Info("Could not read start-param", "err", err)
```

File: cmd/geth/dbcmd.go [hunk 3]
```diff
@@ -537,41 +537,51 @@ func dbDumpTrie(ctx *cli.Context) error {
...
 		return err
 	}
 	if end, err = strconv.ParseInt(ctx.Args().Get(2), 10, 64); err != nil {
-		log.Info("Could read count param", "error", err)
+		log.Info("Could read count param", "err", err)
```
Comment (holiman) [inline cmd/geth/dbcmd.go:559]:
```suggestion
		log.Info("Could not read count param", "err", err)
```

**Review by holiman — APPROVED** [review]
It looks good to me, but it's a bit wonky since it's hardcoded to return `""` subdirs at all times, so a bit difficult to test that it works as it should. I'll trust you though :)

**Comment by rjl493456442** [comment]
Yes, currently there is no sub-directory will be returned. But once we use freezer to store reverse-diff, or store trace-result, then these sub-ancient stores will be registered and corresponding sub-directory will be returned.

**Comment by rjl493456442** [comment]
Close it, since it's reworked in https://github.com/ethereum/go-ethereum/pull/25487/files in a cleaner approach.

**Pull request closed without merging**